premAI-io/prem-services
# add CI build matrix, refactoring, linting

- unify scripts under model dirs
  + fix `cpl-{replit,t5}` vs `cdr-{replit,t5}` typo in build scripts
  + fast-fail build/tests in scripts
  + move `scripts/*_*.sh` -> `*-*/build.sh`
    ```bash
    for f in scripts/*.sh; do
      model=$(basename -s .sh $f | sed s/_/-/g)
      sed -i "s#/$model##g" $f
      git mv $f $model/build.sh
    done
    ```
- add `build.sh --flags` support
- add CI docker build matrix (fixes #49)
  + fix builds (skip GPU tests, check env vars, multi-arch `--load`, etc)
- use self-hosted GPU runners for testing
- fix caching (fixes #50)
  + see e.g. [this build clearly has caching working](https://github.com/premAI-io/prem-services/actions/runs/5666475475/job/15372768862)
- drop CI `pre-commit/action` in favour of [app](https://pre-commit.ci)
- purge unused config & files
- add more pre-commit hooks & flake8 tests
  + fix detected errors
- unify config under `pyproject.toml` <= `setup.cfg`, `pytest.ini`
- open follow-up issues
  * [x] https://github.com/premAI-io/prem-services/issues/56#issuecomment-1651261777
  * [x] #66
  * [x] #67
  * [ ] update online docs
